docs_src/design/adr/security/0017-consul-security.md

```diff
@@ -0,0 +1,111 @@
+# Securing access to Consul
+
+## Status
+
+** proposed **
+
+## Context
+
+This ADR defines the motiviation and approach used to secure access
+to the Consul component in the EdgeX archticture
+for *security-enabled configurations only*.
+(Non-secure configuations continue to use Consul in
+anonymous read-write mode.)
+
+Consul provides several services for the EdgeX architecture:
+
+- Service registry (see ADR in references below)
+- Service health monitoring
+- Mutable configuration data
+
+Use of the services provided by Consul is optional on a service-by-service basis.
+Use of the registry is controlled by the `-r` or `--registry` flag provided to an EdgeX service.
+Use of mutable configuration data is controlled by the `-cp` or `--configProvider` flag provided to an EdgeX service.
+When Consul is enabled as a configuration provider,
+the `configuration.toml` is parsed into individual settings
+and seeded into the Consul key-value store on the first start of a service.
+Configuration reads and writes are then done to Consul if it is specified as the configuration provider,
+otherwise the static `configuration.toml` is used.
+Writes to the `[Writable]` section in Consul trigger per-service callbacks
+notifying the application of the changed data.
+
+Since configuration data can affect the runtime behavior of services,
+compensating controls must be introduced in order to mitigate the risks introduced
+by moving configuration from a static file into to an HTTP-accessible service with mutable state.
+
+The current practice is that Consul is exposed via unencrypted HTTP in anonymous read/write mode
+to all processes and EdgeX services running on the host machine.
+
+## Decision
+
+Consul will be configured with access control list (ACL) functionality enabled,
+and each EdgeX service will utilize a Consul access token to authenticate to Consul.
+Consul access tokens will be requested from the Vault Consul secrets engine
+(to avoid introducing additional bootstrapping secrets).
+
+DNS will be disabled via configuration as it is not used in EdgeX.
+
+**Consul Access Via API Gateway**
+
+In security enabled EdgeX, the API gateway will be configured to
+proxy the Consul service over the `/consul` path,
+using the `request-transformer` plugin
+to add the global management token to incoming requests
+via the `X-Consul-Token` HTTP header.
+Thus, ability to access remote APIs also grants the ability
```

Comment:
which APIs would be referring here? Gateway APIs or Consul's APIs to modify KV for configuration toml values?

Reply:
I am referring to the API gateway feature in aggregate.